problems/problems.go

```diff
@@ -0,0 +1,74 @@
+package errors
+
+import (
+	"fmt"
+)
+
+//Problem implements the RFC7807 "problem"/error standard
+//Additional error field can be defined in custom struct implementing the Problem struct
+type Problem struct {
+	Title    string `json:"title"`
+	Detail   string `json:"detail,omitempty"`
+	Type     string `json:"type,omitempty"`
+	Instance string `json:"instance,omitempty"`
+	Status   int    `json:"status,omitempty"`
+}
+
+//Error implements the error interface
+func (p *Problem) Error() string {
+	return fmt.Sprintf("Title: %s; Detail: %s", p.Title, p.Detail)
+}
+
+//DefaultType implements the default type content described in RFC7807
+const DefaultType = "about:blank"
+
+//SetTitle sets the title of the problem object
+func (p *Problem) SetTitle(title string) *Problem {
+	p.Title = title
+	return p
+}
+
+//SetDetail sets the detail info of the problem object
+func (p *Problem) SetDetail(detail string) *Problem {
+	p.Detail = detail
+	return p
+}
+
+//SetType sets the type of the problem object
+func (p *Problem) SetType(typ string) *Problem {
+	if typ == "" {
+		p.Type = DefaultType
+	} else {
+		p.Type = typ
+	}
+
+	return p
+}
+
+//SetInstance sets the instance of the problem object
+func (p *Problem) SetInstance(instance string) *Problem {
+	p.Instance = instance
+	return p
+}
+
+//SetStatus sets the HTTP status of the problem object HTTP request
+func (p *Problem) SetStatus(status int) *Problem {
+	p.Status = status
+	return p
+}
+
+//New creates a new Problem with all object details described in RFC7807
```

Comment:
New should be first function after it's type definition.

Also are there any mandatory fields (like Title) that should be part of the New() signature?

errors.New(string) is already a well known function.
I would see it as quite intuitive to have problems.New(string) as well if it makes sense

Reply:
Probably a `New(title string)` function would be a good idea. Nevertheless, i would recommend using a function like `New(title, detail string)`, as `title` is likely just the title of the HTTP status code, whereas `detail` describes the problem further.

Reply:
I guess I have to look at the RFC Spec a little more to give a comment on that.

I would wish to make the signature as similar as possible but we'll see. Give me a day or so.